xtask/src/commands/test.rs

```diff
@@ -49,6 +67,15 @@ impl Test {
             Box::new((demo, guard))
         };
 
+        eprintln!("Waiting for service to be ready...");
+        loop {
+            match reqwest::blocking::get("http://localhost:4100/graphql") {
+                Ok(_) => break,
+                Err(err) => eprintln!("{}", err),
+            }
+            sleep(Duration::from_secs(2));
+        }
```

Comment:
I assume you moved this here to make sure the federation-demo is running before continuing even if it's handled by the user?

Reply:
I moved this here so we can spawn a cargo test --no-run while we run npm i and npm run start. This allows us to speedup the first test build time, well it technically doesn't speed up the build time, it just starts earlier.